src/lp_data/Highs.cpp

```diff
@@ -2431,6 +2433,20 @@ HighsStatus Highs::getColName(const HighsInt col, std::string& name) const {
   return HighsStatus::kOk;
 }
 
+HighsStatus Highs::getColByName(const std::string& name, HighsInt& col) {
+  HighsLp& lp = model_.lp_;
+  if (!lp.col_names_.size()) return HighsStatus::kError;
+  if (!lp.name2col_.size()) {
+    for (HighsInt iCol = 0; iCol < lp.num_col_; iCol++)
+      lp.name2col_.emplace(lp.col_names_[iCol], iCol);
```

Comment:
So I guess this has the same behavior as Gurobi: https://www.gurobi.com/documentation/10.0/refman/c_getvarbyname.html

> Retrieves a variable from its name. If multiple variables have the same name, this routine chooses one arbitrarily.

Reply:
Other than having vectors of length `num_col` and `num_row` indicating whether a name is duplicate, is there a neat way to know whether duplicates exist?

Reply:
When creating `lp.name2col_`, if you detect a key is already added, then change the value to some sentinel value like `-1`. 

Then, on query, if the value is `-1`, throw an error.

Reply:
Because it's okay for multiple rows/columns to have duplicate names (e.g., during printing). The problem is only if someone tries to query them.

Not sure if you check for duplicate names when writing out LP/MPS files, but you need to take that into account too.

Reply:
> When creating lp.name2col_, if you detect a key is already added, then change the value to some sentinel value like -1.

That's neat, thanks

We don't currently check for duplicate names when writing out LP/MPS files - just empty column/row name vectors - but I realised that this would allow us to do so. Would you throw an error, or just a warning?

Reply:
On files, it's a bit involved, but we deduplicate names by appending `_i` to duplicate names, where `i` increments every time we see the same name:
https://github.com/jump-dev/MathOptInterface.jl/blob/539f1aa47c1805023502644b643e84aeb3d18435/src/FileFormats/utils.jl#L11-L178

For now you could just leave it, and rely on the consumer of the file to detect duplicate names. It's the user's fault.